### PR TITLE
Reduce ready index consume overhead

### PR DESCRIPTION
**Pull request by Ayan-sh03**
Summary:
- optimize ready-index claim to avoid ready-value reads on fresh messages and disable unnecessary ready iterator value prefetch
- keep fallback for requeued ready pointers whose ready sequence differs from original message sequence
- update cmd/bench kueue ack requests to include deliveryToken

Tests:
- go test -count=1 ./...
- go build ./...
- smoke benchmark: go run ./cmd/bench -targets kueue -messages 200 -warmup 20 -runs 1 -payload-bytes 256 -producers 1 -consumers 1 -prefetch 50

## Summary by CodeRabbit

* **New Features**
  * Acknowledgment requests now include a delivery token from receive responses to improve message tracking.

* **Improvements**
  * Refined ack-rate calculation for more accurate metrics reporting.

* **Tests**
  * Added unit tests validating acknowledgment payloads, key parsing behavior, and ack-rate computation across edge cases.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -542,36 +551,46 @@ func claimNextReadyMessage(queueId string) (*Message, error) {
...
+					})
+					if valueErr != nil {
+						return fmt.Errorf("parse ready value: %w", valueErr)
+					}
+					msgKey = messageKeyBytes(queueId, originalSeq, msgID)
+					msgItem, err = txn.Get(msgKey)
+					if err == badger.ErrKeyNotFound {
+						txn.Delete(rKey)
+						continue
+					}
```
[the remark this answers is not in the document]
Comment (Ayan-sh03) [reply main.go:591]: done
